save_awq.py

```diff
@@ -0,0 +1,101 @@
+"""Converts and store AWQ-quantized model."""
+
+import argparse
+import json
+import logging
+
+import awq
+import transformers
+
+QUANT_CONFIG = {
+    "zero_point": True,
+    "q_group_size": 128,
+    "w_bit": 4,
+    "version": "GEMM",
+}
+
+
+def parse_args():
+    """Parses command line arguments."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--dataset",
+        type=str,
+        default="msp_open_ai_ada2_random_s5000_gpt4_da0_mr20_sampled_mix.jsonl",
+        help="Path to the calibration dataset.",
+    )
+    parser.add_argument(
+        "--model_path",
+        type=str,
+        default="castorini/rank_zephyr_7b_v1_full",
+        help="Path/slug to the original model.",
+    )
+    parser.add_argument(
+        "--quant_path",
+        type=str,
+        default="awq_rank_zephyr_7b_v1_full",
+        help="Path/slug where the quantized model is to be stored.",
+    )
+    args = parser.parse_args()
+    return args
+
+
+def load_dataset(dataset: str):
+    """Returns list of prompts for given dataset."""
+    with open(dataset, "r") as file:
+        data = json.load(file)
+    prompts = []
+    for content in data:
+        content = content["conversations"]
+        prompt = ""
+        for prompt_dict in content:
+            if prompt_dict["from"] == "system":
```

Comment:
looks like you are appending all the system parts first, then human, then gpt into the prompt? Why are you separating them, rather than just loading the json dataset as is?

Reply:
I followed the example given in the official documentation: https://github.com/casper-hansen/AutoAWQ/blob/main/docs/examples.md#custom-data.

Reply:
Thank you for the pointer code, my understanding from it is that for each set of "instruction", "context" and "response" it concatenates them into string of "instruction"+"response"+"context". However it seems like your code will generate a list of strings like "system"+"system"+...+"system"+"human"+human"+"human"+..+"human"+"gpt"+"gpt"+...+"gpt" which is not quite the same

Reply:
I see what you mean, thanks for the explanation, but it's not what happening here. This JSON has around 39K conversations and each conversation has list of three dictionaries with `system`, `human`, and `gpt` as the value of `from` key. So, for a given dictionary (i.e. for a given conversation), I first extract the system prompt then human and finally the gpt one. Therefore, the `prompt` variable at line 60 will hold `\n` separated values of these fields and each `prompt` is appended to final list `prompts`.

Reply:
I see, thanks for explaining